ascent/examples/sequences_using_recursive_records.rs

```diff
@@ -0,0 +1,79 @@
+//! Sequences using Recursive Records
```

Comment:
While I agree with it being mostly about showcasing ADTs in Soufflé I think it can be useful for outlining a possible way of modeling dynamic lists (or persistent data structures in general) in ascent. It wasn't immediately obvious to me that this kind of stuff would work, even though in hindsight it totally makes sense.